### DISCUSSION TITLE
How to pass data from one route to another in Nextjs App routes?

### DISCUSSION THREAD

**Discussion by marif0025** [vercel/next.js#48439] [Help]
### Summary

I have recently transitioned from the pages directory to the app directory, and I am seeking assistance in sharing data between pages. In the pages directory, I employed the `router.push()` method to transmit data in query parameters. Additionally, I used the `as` attribute to display only the pathname in the address bar for a cleaner URL. However, in the app directory, this functionality is not functioning as expected.

As a result, I must revert to using the pages directory for this particular use case. I would appreciate any assistance you can provide. Thank you.

**Comment by SamKomesarook** [comment]
Are you able to use [route params](https://nextjs.org/docs/app/api-reference/functions/use-search-params), or do you prefer the cleaner URL? If the latter, perhaps consider some form of state management, like a [context](https://react.dev/reference/react/useContext).

**Reply by marif0025** [reply]
No, I prefer cleaner URL and use a hack to do the job. Actually that works even better.

**Reply by SamKomesarook** [reply]
Perhaps then close the discussion?

**Reply by marif0025** [reply]
Yes. Offcourse.
